data/plugins/treeview.lua

```diff
@@ -445,14 +454,22 @@ function TreeView:draw()
   local doc = core.active_view.doc
   local active_filename = doc and system.absolute_path(doc.filename or "")
 
+  local ox = math.abs(self:get_content_offset())
+  local sw, ox = 0, math.abs(self:get_content_offset())
   for item, x,y,w,h in self:each_item() do
     if y + h >= _y and y < _y + _h then
-      self:draw_item(item,
+      local w = self:draw_item(
+        item,
         item == self.selected_item,
         item == self.hovered_item,
-        x, y, w, h)
+        x, y, w, h
+      ) + ox
+      sw = math.max(w, sw)
     end
   end
+  if sw > self.scroll.x + self.size.x then
+    self.scroll_width = sw
+  end
```

Comment:
> We could even keep track of whether the user is currently drag-scrolling (self.h_scrollbar.dragging), wait until they're done, then reduce the scrollable size.

I've thought about this, but I think this might introduce an issue if the user is scrolling with the wheel, since that counts as individual scrolling events.